### PR TITLE
bpo-42393: Raise OverflowError iso. DeprecationWarning on overflow in socket.ntohs and socket.htons

### PR DESCRIPTION
**Pull request by erlend-aasland**
https://bugs.python.org/issue42393

### CODE REVIEW

**Comment by erlend-aasland** [comment]
@serhiy-storchaka, would you mind reviewing this?

File: Modules/socketmodule.c [hunk 1]
```diff
@@ -6117,11 +6114,8 @@ PyDoc_STRVAR(ntohs_doc,
 "ntohs(integer) -> integer\n\
 \n\
 Convert a 16-bit unsigned integer from network to host byte order.\n\
-Note that in case the received integer does not fit in 16-bit unsigned\n\
-integer, but does fit in a positive C int, it is silently truncated to\n\
-16-bit unsigned integer.\n\
-However, this silent truncation feature is deprecated, and will raise an\n\
-exception in future versions of Python.");
+An OverflowError is raised if the received integer does not fit in\n\
```
Comment (serhiy-storchaka) [inline Modules/socketmodule.c]: OverflowError is usually not mentioned specially in docstrings. See for example docstrings for ntohl/htonl.

File: Doc/library/socket.rst [hunk 2]
```diff
@@ -925,13 +920,8 @@ The :mod:`socket` module also offers various network-related services:
 
    Convert 16-bit positive integers from host to network byte order.  On machines
    where the host byte order is the same as network byte order, this is a no-op;
-   otherwise, it performs a 2-byte swap operation.
-
-   .. deprecated:: 3.7
```
Comment (serhiy-storchaka) [inline Doc/library/socket.rst:930]: Add `versionchanged` instead of `deprecated`.

File: Doc/library/socket.rst [hunk 3]
```diff
@@ -925,13 +920,8 @@ The :mod:`socket` module also offers various network-related services:
...
-      In case *x* does not fit in 16-bit unsigned integer, but does fit in a
-      positive C int, it is silently truncated to 16-bit unsigned integer.
-      This silent truncation feature is deprecated, and will raise an
-      exception in future versions of Python.
+   otherwise, it performs a 2-byte swap operation.  :exc:`OverflowError` is raised
```
Comment (serhiy-storchaka) [inline Doc/library/socket.rst]: There is no special mentioning of OverflowError in the documentation of htonl/ntohl, and I think that there is no need to add it here.

**Comment by erlend-aasland** [comment]
Thank you for reviewing, @serhiy-storchaka. I'll push the changes when I get to my laptop later today.

**Review by serhiy-storchaka — APPROVED** [review]

**Pull request merged**